lib/tpm/tpm_simulator_test.go

```diff
@@ -0,0 +1,230 @@
+//go:build tpmsimulator
+
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package tpm
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"log/slog"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/google/go-attestation/attest"
+	tpmsimulator "github.com/google/go-tpm-tools/simulator"
+	"github.com/google/go-tpm/legacy/tpm2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/fixtures"
+)
+
+// fakeCmdChannel is used to inject the TPM simulator into `go-attestation`'s
+// TPM wrapper.
+type fakeCmdChannel struct {
+	io.ReadWriteCloser
+}
+
+// MeasurementLog implements CommandChannelTPM20.
+func (cc *fakeCmdChannel) MeasurementLog() ([]byte, error) {
+	// Nothing to do here - we don't use the measurement log for tpm joining.
+	return nil, nil
+}
+
+func writeEKCertToTPM(t *testing.T, sim *tpmsimulator.Simulator, data []byte) {
+	const nvramRSAEKCertIndex = 0x1c00002
```

Comment:
Is this arbitrary or does it have to be this specific index?

Reply:
Yeah - this is the TCG defined index for the RSA EKCert. I can add words to that effect as a comment.

Reply:
e996c1e397